### PR TITLE
Feature/dwr 1156 audit it

### PR DESCRIPTION
**Pull request by chdietz**
* Test TRT's from Audit.md
* mock package, setup script and submit script
* setup sql with same data as setup script
* IT test to process TRT's and check counts

### CODE REVIEW

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultTDSReportProcessorAuditIT.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+     * SSID004  (ethnicity=1          sum=1           )  Student-update=0   exam=2  update=0
+     * -----------------------------------------------------------------------------------------
+     * Student=4(                     sum=8  Deleted=3)  Student-update=7   exam=9  update=6
+     */
+    @Test
+    public void itShouldProcessExamsAndAuditChanges() {
+
+        final int importId = -1;
+        final String[] testResults = {
+                "B001-SSID001-IAB-010-V01.xml",
```
Comment (agorina) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultTDSReportProcessorAuditIT.java]: Is there a pattern in the naming convention of the files? If yes, would be nice to note it here (unless it is already done and I just missed it).

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultTDSReportProcessorAuditIT.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                "B004-SSID001-ICA-030-V03.xml",
+                "B004-SSID002-IAB-040-V02.xml",
+                "B005-SSID001-ICA-030-V04.xml",
+                "B006-SSID001-ICA-060-V01.xml",
+                "B006-SSID003-ICA-050-V02.xml",
+                "B006-SSID004-IAB-070-V01.xml",
+                "B007-SSID001-IAB-020-V01.xml",
+                "B007-SSID004-ICA-080-V01.xml"};
+
+        assertThat(countRowsInTable(jdbcTemplate, "asmt")).isEqualTo(3);
```
Comment (agorina) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultTDSReportProcessorAuditIT.java]: Nit: all other ITs count rows before and then compare the count increment. It is more forgiving in a case where there are some data in the tables.

File: exam-processor/src/test/resources/examStudentAudit/PreloadSetupData.sql [hunk 3]
```diff
@@ -0,0 +1,54 @@
+-- Ensure auditing is on.
+UPDATE setting s SET s.value = 'TRUE' WHERE s.name = 'AUDIT_TRIGGER_ENABLE';
+
+-- Create with mysqldump after setup_data.sh on a cleaned and migrated warehouse
+-- Example:
+-- mysqldump --user=root --host=127.0.0.1 --default-character-set=utf8 --no-create-info=TRUE --compact --complete-insert --skip-quote-names --result-file=dump.sql warehouse import asmt item district school accommodation
+-- remove import id of 1 as it is added by flyway
+-- add import id of -1 used by integration tests
```
Comment (agorina) [inline exam-processor/src/test/resources/examStudentAudit/PreloadSetupData.sql]: Are above comments applicable here? Looks like a leftover from something else.

Comment (chdietz) [reply exam-processor/src/test/resources/examStudentAudit/PreloadSetupData.sql]: Yes, these are instructions on how to regenerate this file.  The simple fact you did not get that mean I need to edit them a bit ;)

The idea is to make it easy to keep this suite of test files easy to maintain.

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultTDSReportProcessorAuditIT.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    private TDSReportProcessor processor;
+
+    /**
+     * The same test results used for example query output in  See <a href="http://github.com/SmarterApp/RDW/blob/develop/docs/Audit.md">Audit.md</a> are
+     * used here for an integration test.  In addition to running as an integration test setup and submit scripts are provided to process into
+     * a running ingest.
+     * <p>
+     * The objective of this test validating expected updates and auditing.
+     * File format: Batch-SSID-TestType-OppId-UpdateVersion.
+     * Batch is for submission order.  Files with the same batch could process in any order in the logical batch.
```
Comment (chdietz) [inline exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultTDSReportProcessorAuditIT.java]: File naming convention for TRT's

File: exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultTDSReportProcessorAuditIT.java (hunk 1 above)
Comment (chdietz) [reply exam-processor/src/test/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultTDSReportProcessorAuditIT.java]: Yes there is above.  There are a few snippets of information spread between the IT, the .sql file and the bash scripts perhaps I will add a overview readme that explains what all the different file are for in this folder.

**Pull request merged**